mixedprecision/roots/rpoly.cpp

```diff
@@ -0,0 +1,1052 @@
+// rpoly.cpp: mixed-precision experiments with Rpoly root finding method
+//
+// Copyright (C) 2017 Stillwater Supercomputing, Inc.
+// SPDX-License-Identifier: MIT
+//
+// This file is part of the universal numbers project, which is released under an MIT Open Source license.
+// Portions of this file are adapted from work by Chris Sweeney (2015) under the 3-Clause BSD License.
+// See https://github.com/sweeneychris/RpolyPlusPlus for more details
+
+#include <iostream>
+#include <iomanip>
+
+#include <universal/number/posit/posit.hpp>
+#include <universal/number/cfloat/cfloat.hpp>
+
+#include <universal/blas/blas.hpp>
+
+template <typename Real>
+sw::universal::blas::vector<Real> Head(const sw::universal::blas::vector<Real>& vector, size_t size){
+    sw::universal::blas::vector<Real> head(size);
+    for (size_t i = 0; i < size && i < vector.size(); ++i){
+        head[i] = vector[i];
+    }
+
+    return head;
+}
+
+template <typename Real>
+sw::universal::blas::vector<Real> Tail(const sw::universal::blas::vector<Real>& vector, size_t size){
+    sw::universal::blas::vector<Real> tail(size);
+    size_t startIdx = vector.size() - size > 0 ? vector.size() - size : 0;
+    for (size_t i = startIdx; i < vector.size(); ++i){
+        tail[i] = vector[i - startIdx];
+    }
+
+    return tail;
+}
+
+template <typename Real>
+void SetZero(sw::universal::blas::vector<Real>& vector, size_t size){
```

Comment:
this semantic is in the library

using Real = posit<32,2>;
using Vector = blas::vector<Real>;

Vector v(7341, 0);  // construct size and set elements to value

Vector w(16);

w = Real(0);  // this will broadcast the value across the vector elements